app/validation/CsvLineValidator.scala

```diff
@@ -38,12 +38,12 @@ trait FieldValidator {
     }
   }
 
-  def validateNino(nino: String) = {
+  def validateNino(nino: String): Option[String] = {
     val TEMP_NINO = "TN"
     nino match {
       case "" => Some(Messages("gmp.error.mandatory", Messages("gmp.nino")))
-      case x if !NinoValidate.isValid(x) => Some(Messages("gmp.error.nino.invalid"))
       case x if x.toUpperCase().startsWith(TEMP_NINO) => Some(Messages("gmp.error.nino.temporary"))
```

Comment:
Reordered the pattern matching so that temporary NINOs are matched before the validation takes place.